### PR TITLE
Add database name and description property

### PR DESCRIPTION
**Pull request by janbrummer**

### CODE REVIEW

**Comment by janbrummer** [comment]
@Evidlo Could you please review this one?

**Comment by janbrummer** [comment]
https://gitlab.gnome.org/World/secrets/-/merge_requests/878

**Comment by janbrummer** [comment]
@A6GibKm Time for a review?

File: pykeepass/pykeepass.py [hunk 1]
```diff
@@ -268,6 +268,42 @@ def entries(self):
         excluding history"""
         return self.find_entries()
 
+    @property
+    def database_name(self):
+        """Name of database"""
+        elem = self._xpath('/KeePassFile/Meta/DatabaseName', first=True)
```
Comment (A6GibKm) [inline pykeepass/pykeepass.py:274]: I see that in the other cases the `path` is inlined in the `_xpath` fn I would do them all the same, and if there is already a code style in this file, follow it.

File: pykeepass/pykeepass.py [hunk 2]
```diff
@@ -268,6 +268,42 @@ def entries(self):
...
+
+    @database_name.setter
+    def database_name(self, name):
+        path = '/KeePassFile/Meta/DatabaseName'
+        item = self._xpath(path, first=True)
+        item.text = str(name)
+
+    @property
+    def database_description(self):
+        """Description of database"""
```
Comment (A6GibKm) [inline pykeepass/pykeepass.py:285]: of *the* database

File: pykeepass/pykeepass.py [hunk 3]
```diff
@@ -268,6 +268,42 @@ def entries(self):
...
+    def database_description(self, name):
+        path = '/KeePassFile/Meta/DatabaseDescription'
+        item = self._xpath(path, first=True)
+        item.text = str(name)
+
+    @property
+    def default_username(self):
+        """Default Username"""
+        elem = self._xpath('/KeePassFile/Meta/DefaultUserName', first=True)
+        return elem.text
```
Comment (A6GibKm) [inline pykeepass/pykeepass.py:299]: can this return None? or will it return `""`? This should be documented.

**Review by A6GibKm — COMMENTED** [review]
Thanks. Please add a test for the setters and getters. 

- get value
- set value to X
- get value and check that it matches X
- save db
- close db
- open db
- check the it is still X

**Comment by A6GibKm** [comment]
Is this missing a rebase? I think python 3.6 tests were removed one or two weeks ago.

File: pykeepass/pykeepass.py (hunk 1 above)
Comment (janbrummer) [reply pykeepass/pykeepass.py:274]: I've used recyclebin_group as base, so this is existing code style. What should i change then? There is no Coding Guidelines in this project..

File: pykeepass/pykeepass.py (hunk 2 above)
Comment (janbrummer) [reply pykeepass/pykeepass.py:285]: Updated

File: pykeepass/pykeepass.py (hunk 3 above)
Comment (janbrummer) [reply pykeepass/pykeepass.py:299]: In case it is not set it is None. Added description.

File: pykeepass/pykeepass.py (hunk 1 above)
Comment (A6GibKm) [reply pykeepass/pykeepass.py:274]: If the file is consistent in its style follow that, if not choose one style of your liking and use the same for all 4 fns.

Comment (janbrummer) [reply pykeepass/pykeepass.py:274]: Done

**Comment by janbrummer** [comment]
> Thanks. Please add a test for the setters and getters.
> 
>     * get value
> 
>     * set value to X
> 
>     * get value and check that it matches X
> 
>     * save db
> 
>     * close db
> 
>     * open db
> 
>     * check the it is still X

Added tests

**Comment by janbrummer** [comment]
@Evidlo Can you please merge it?

**Pull request merged**